core/src/main/scala/loadbalancers/Aws.scala

```diff
@@ -66,28 +66,33 @@ final class Aws(cfg: Infrastructure.Aws) extends (LoadbalancerOp ~> IO) {
 
   def delete(lb: Datacenter.LoadbalancerDeployment): IO[Unit] = {
     val name = loadbalancerName(lb.loadbalancer.name, lb.loadbalancer.version, lb.hash)
-    for {
+    (for {
       lbarn <- getLoadBalancerArn(name)
-      _ <- deleteNLB(lbarn)
-
       targetGroups <- getTargetGroupsForNLB(lbarn)
+      _ <- deleteNLB(lbarn)
       _ <- deleteTargetGroups(targetGroups)
 
       _ <- deleteASG(name)
-    } yield ()
+    } yield ()).recoverWith{
+      case e: Exception =>
+        log.info(s"couldn't delete NLB because ${e.getMessage}. attempting to delete ELB")
+        for {
+          _ <- deleteELB(name)
```

Comment:
What happens if these fail?

Reply:
Errors will get logged, but Nelson will still mark the resources as terminated. Not much we can do :man_shrugging: